### PR TITLE
route.go: allow to chain instrumentations

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by roidelapluie** [comment]
The purpose is to clean https://github.com/prometheus/prometheus/blob/64194f7d45cbb7fa2f41c8f219706308f65d511a/web/web.go#L1121-L1129

File: route/route.go [hunk 1]
```diff
@@ -53,6 +53,12 @@ func New() *Router {
 
 // WithInstrumentation returns a router with instrumentation support.
 func (r *Router) WithInstrumentation(instrh func(handlerName string, handler http.HandlerFunc) http.HandlerFunc) *Router {
+	if r.instrh != nil {
```
Comment (brian-brazil) [inline route/route.go]: Why would it be nil?

Comment (roidelapluie) [reply route/route.go]: It is the zero value for a function

Comment (brian-brazil) [reply route/route.go]: That doesn't answer my question. Where are we calling this with nil, and why?

**Comment by roidelapluie** [comment]
We check the existing instrumentation (r.ins) not the new one.

Le sam. 11 janv. 2020 à 10:10, Brian Brazil <notifications@github.com> a
écrit :

File: route/route.go (hunk 1 above)
Comment (roidelapluie) [reply route/route.go]: When we call router.New(). This is the existing instrumentation.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**